### ISSUE TITLE
Istio RBAC rules are not working for `Origin Authentication` when `originIsOptional: true`

### ISSUE THREAD

**Issue by yskopets** [istio/proxy#1958]
#### Description

I'm trying to create a configuration with RBAC rules for both `service-2-service` and `user-2-service` traffic at the same time.

Let's say I have service `A`. I want to allow requests in 2 cases: 
  1) requests are coming from service `B` (in which case `Origin Authentication` should not be enforced) 
  2) requests are coming from humans via ingress gateway (in which case `Origin Authentication` must be enforced)

I'm running into an issue that it's not possible to achieve with `Istio 1.0.0/1.0.1`:
- when I set `originIsOptional: false`, I get RBAC working for `user-2-service` traffic; but `service-2-service` traffic is broken because it's missing JWT token mandatory for `Origin Authentication`
- when I set `originIsOptional: true`, I get RBAC working for `service-2-service` traffic; but `user-2-service` traffic is broken because JWT token is no longer taken into account (even though it's present in the request)

The reason for that is the source code of `istio_authn` filter (https://github.com/istio/proxy/blob/1.0.1/src/envoy/http/authn/http_filter.cc#L62)

```
  bool success =
      filter_config_.policy().origin_is_optional() ||
      createOriginAuthenticator(filter_context_.get())->run(&payload);
```

when `originIsOptional: true`, then `istio_authn` filter doesn't even try to handle JWT Token extracted by `jwt-auth`; that's why info from the JWT Token is not available to `envoy.filters.http.rbac` filter

#### Expected behaviour

It should possible to have RBAC rules working for both `service-2-service` and `user-2-service` traffic at the same time.

For example, when `originIsOptional: true`:
1. if a request is missing a JWT Token then `istio_authn` filter should continue request processing without passing `Origin Authentication` info to the `envoy.filters.http.rbac` filter
2. if a request includes a JWT Token then `istio_authn` filter should continue request processing and pass`Origin Authentication` info to the `envoy.filters.http.rbac` filter

#### Configuration to reproduce the issue

```
---
apiVersion: authentication.istio.io/v1alpha1
kind: Policy
metadata:
  name: httpbin-policy
  namespace: example-app-1
spec:
  targets:
  - name: httpbin
    ports:
    - number: 8000
  peers:
  - mtls: {}
  originIsOptional: true
  origins:
  - jwt:
      issuer: "https://example.org"
      jwksUri: "https://example.org/keys"
      jwt_headers:
      - "x-example-jwt"
  principalBinding: USE_PEER
---
apiVersion: "rbac.istio.io/v1alpha1"
kind: ServiceRole
metadata:
  name: admin
  namespace: example-app-1
spec:
  rules:
  - services: ["*"]
    methods: ["*"]
---
apiVersion: "rbac.istio.io/v1alpha1"
kind: ServiceRoleBinding
metadata:
  name: admin-binding
  namespace: example-app-1
spec:
  subjects:
  # service-2-service scenario
  - properties:
      source.namespace: "example-app-2"
  # user-2-service scenario (via Ingress Gateway)
  - properties:
      request.auth.claims[user]: "admin" # this info is not available to `envoy.filters.http.rbac` filter when `originIsOptional: true`
      source.namespace: "istio-system" # Ingress Gateway
  roleRef:
    kind: ServiceRole
    name: "admin"
```

#### Affected version

Istio v1.0.0, v1.0.1

**Comment by yangminzhu** [comment]
Hi Yaroslav,

Sorry for the issue and thanks for your detail report. It's a known issue,
see https://github.com/istio/proxy/issues/1892.

We initially thought the `originIsOptional` flag is causing confusion and
could be replaced with the per-path JWT authentication
<https://docs.google.com/document/d/193HcSBSQOAs0zFBbemOd9vis3s5mmsrtLJBc99XfS7o/edit>.
But your case seems a reasonable use of `originIsOptional` flag, Maybe we
should re-evaluate this and fix the bug and let the user to use it if they
have such a custom use case? +Diem Vu <diemvu@google.com>  +Limin Wang
<liminwang@google.com>

Yangmin
---

**Comment by yskopets** [comment]
Hi Yangmin!

Thanks for the feedback!

I’d like to run by you another usage scenario.

In our existing Kubernetes setup (without Istio):
- we enforce `User-2-Service` security policies at `Ingress` level
- and use `Prometheus Operator` to scrape metrics from all applications
- since Prometheus scrapes metrics directly from `Pods`, it is not affected by our `User-2-Service` security solution

For comparison, in Istio case:

- `User-2-Service` security is enforced at `Pod` level
- and due to the issue with `originIsOptional: true`, Prometheus will be denied to scrape the app unless it presents a token for `Origin Authentication`

How do you solve this problem ? (a combination of `Origin Authentication` and Prometheus scrapes)

While HTTP metrics could be collected exclusively by Envoy/Mixer, application-specific metrics still need to be scraped from the app itself.

**Comment by yskopets** [comment]
Hi!

Yes, I confirm that the issue is fixed as of `Istio v1.0.5`
